src/utils/transaction/insertTransaction.ts

```diff
@@ -15,141 +16,225 @@ import { reportUsage } from "../usage";
-): Promise<string> => {
-  const { insertedTransaction, idempotencyKey, shouldSimulate = false } = args;
+const transformV5SmartBackendWallet = async (
+  transaction: QueuedTransaction,
+  walletDetails: SmartBackendWalletDetails, // Note: narrowed type
+): Promise<QueuedTransaction> => {
+  await validateSmartBackendWalletChainSupport(transaction.chainId);
 
-  // The queueId uniquely represents an enqueued transaction.
-  // It's also used as the idempotency key (default = no idempotence).
-  let queueId: string = randomUUID();
-  if (idempotencyKey) {
-    queueId = idempotencyKey;
-    if (await TransactionDB.exists(queueId)) {
-      // No-op. Return the existing queueId.
-      return queueId;
-    }
+  if (transaction.accountAddress) {
+    throw createCustomError(
+      "Smart backend wallets do not support interacting with other smart accounts",
+      StatusCodes.BAD_REQUEST,
+      "INVALID_SMART_BACKEND_WALLET_INTERACTION",
+    );
   }
 
-  let queuedTransaction: QueuedTransaction = {
-    ...insertedTransaction,
-    status: "queued",
-    queueId,
-    queuedAt: new Date(),
-    resendCount: 0,
-
-    from: getChecksumAddress(insertedTransaction.from),
-    to: getChecksumAddress(insertedTransaction.to),
-    signerAddress: getChecksumAddress(insertedTransaction.signerAddress),
-    accountAddress: getChecksumAddress(insertedTransaction.accountAddress),
-    accountSalt: insertedTransaction.accountSalt,
-    target: getChecksumAddress(insertedTransaction.target),
-    sender: getChecksumAddress(insertedTransaction.sender),
-    value: insertedTransaction.value ?? 0n,
+  return {
+    ...transaction,
+    isUserOp: true,
+    signerAddress: walletDetails.accountSignerAddress,
+    from: walletDetails.accountSignerAddress,
+    accountAddress: transaction.from, // Original 'from' becomes the account
+    target: transaction.to,
+    accountFactoryAddress: walletDetails.accountFactoryAddress ?? undefined,
+    entrypointAddress: walletDetails.entrypointAddress ?? undefined,
+    walletType: walletDetails.type,
   };
+};
+
+/**
+ * Transform transaction for Case 2 (V4 Smart Backend Wallet)
+ * Type guard ensures walletDetails has required smart wallet properties
+ */
+const transformV4SmartBackendWallet = async (
+  transaction: QueuedTransaction,
+  walletDetails: SmartBackendWalletDetails, // Note: narrowed type
+): Promise<QueuedTransaction> => {
+  await validateSmartBackendWalletChainSupport(transaction.chainId);
 
-  let walletDetails: ParsedWalletDetails | undefined;
+  return {
+    ...transaction,
+    entrypointAddress: walletDetails.entrypointAddress ?? undefined,
+    accountFactoryAddress: walletDetails.accountFactoryAddress ?? undefined,
+    walletType: walletDetails.type,
+  };
+};
 
+/**
+ * Try to resolve wallet details, determining if we're in V4 or V5 case
+ * For V5: wallet details should be found from 'from' address (Cases 1 & 3)
+ * For V4: wallet details are found from accountAddress (Cases 2 & 4)
```

Comment:
This is kinda weird for case 4: v4 SDK for an EOA sender. This flow sets the `accountAddress` even if it's not a userOp?